src/main/scala/devices/debug/Periphery.scala

```diff
@@ -51,77 +51,89 @@ class DebugIO(implicit val p: Parameters) extends ParameterizedBundle()(p) with
   val disableDebug = p(ExportDebug).externalDisable.option(Bool(INPUT))
 }
 
+class PSDIO(implicit val p: Parameters) extends ParameterizedBundle()(p) with CanHavePSDTestModeIO {
```

Comment:
`ParameterizedBundle` is now unnecessary if all the parameters of the class are `val`s (as is the case here). So you can just extend `Bundle`

Reply:
done